File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String[] str = Integer.toString(n, k).split("0");
        for (int i = 0; i < str.length; i++) {
            if (!str[i].isEmpty() && prime(Long.parseLong(str[i]))) {
                answer++;
            }
        }
        return answer; 
    }
    
    public boolean prime(long n) {
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false; 
            }
        }
        return n > 1;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def solution(n, k):
    answer = 0
    k_num = k_number_converter(n, k)
    k_num_zero_split_list = k_num.split("0")
    
    for num in k_num_zero_split_list:
        if num == '':
            continue
        
        if is_prime(int(num)):
            answer += 1
    
    return answer

# 소수 판별 함수
def is_prime(n):
    if n < 2:
        return False
    for i in range (2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

# k 진법 변환 함수
def k_number_converter(n, k):
    # StringBuilder
    digits = []
    while n > 0:
        n, mod = divmod(n, k)
        digits.append(str(mod))
    return ''.join(reversed(digits))